### PR TITLE
Introduced Continuous Intgration

### PR DESCRIPTION
**Pull request by ohbus**
# Leveraging GitHub Actions introduced two pipelines

</br>

- [x] **Signal Server CI :** for running on the master branch

</br>

- [x] **Signal Server Pull Request Builder :** for running Pull Request jobs

### CODE REVIEW

**Review by gautamkrishnar — CHANGES_REQUESTED** [review]
You dont have to put the PR action and push action on seprate files. Since both action's code are the same. 

Simply add the following to .github/workflow/mvn-ci.yml:
```
on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]
    types: [ opened, reopened ]
```

Delete the .github/workflow/mvn-pr.yml

File: .github/workflow/mvn-pr.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Signal Server PR Builder
+
+on:
+  pull_request:
+    branches: [ master ]
+    types: [ opened, reopened, synchronize, labeled, unlabeled ]
```
Comment (gautamkrishnar) [inline .github/workflow/mvn-pr.yml:6]: I think just using `[ opened, reopened ]` would be enough

**Comment by ohbus** [comment]
> You dont have to put the PR action and push action on seprate files. Since both action's code are the same.
> 
> Simply add the following to .github/workflow/mvn-ci.yml:
> 
> ```
> on:
>   push:
>     branches: [ master ]
>   pull_request:
>     branches: [ master ]
>     types: [ opened, reopened ]
> ```
> 
> Delete the .github/workflow/mvn-pr.yml

I have added it for modularity as in the **`master`** branch there will be integrations with sonar and other code coverage tools.

File: .github/workflow/mvn-pr.yml (hunk 1 above)
Comment (ohbus) [reply .github/workflow/mvn-pr.yml:6]: labeled and unlabeled triggers can be removed.

**Comment by gautamkrishnar** [comment]
> I have added it for modularity as in the **`master`** branch there will be integrations with sonar and other code coverage tools.

👍🏻 then its ok,

File: .github/workflow/mvn-pr.yml (hunk 1 above)
Comment (gautamkrishnar) [reply .github/workflow/mvn-pr.yml]: 👍🏻

**Review by gautamkrishnar — APPROVED** [review]

**Comment by alerque** [comment]
> I have added it for modularity as in the **`master`** branch there will be integrations with sonar and other code coverage tools.

This is sideways and will make for lots of duplication and inconsistent results. The molecularity should go the other way. Have a workflow file for **type of job**, not one per trigger. Put all your code coverage jobs in their own workflow file with the appropriate trigger list. Put build jobs in another. Linters in another. Etc.

**Comment by ehrenkret-signal** [comment]
Hmm; I'm not sure what's material different about this versus the .github/workflows/test.yml file that's currently in the repo. Additionally, that file used to run on pull requests but was changed to push only.

**Pull request closed without merging**